File: product-service/src/main/java/com/aleksander/storefront/productservice/domain/repository/ProductRepository.java
package com.aleksander.storefront.productservice.domain.repository;

import com.aleksander.storefront.productservice.context.ProductId;
import com.aleksander.storefront.productservice.domain.Product;

import java.util.List;

public interface ProductRepository {
    Product save(Product product);

    List<Product> getAll();
    List<Product> getAllByIds(List<ProductId> ids);
}

File: order-service/src/main/kotlin/com/aleksander/storefront/orderservice/domain/repository/ProductRepository.kt
package com.aleksander.storefront.orderservice.domain.repository

import com.aleksander.storefront.orderservice.context.ProductId
import com.aleksander.storefront.orderservice.domain.CountryCode
import com.aleksander.storefront.orderservice.domain.CurrencyCode
import com.aleksander.storefront.orderservice.domain.Product

interface ProductRepository {
    fun getAllByIdsAndCountry(ids: List<ProductId>, country: CountryCode): List<Product>
}